### PR TITLE
add format task for vscode

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: always need run this before a push, so maybe it's good to add it here.
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -32,6 +32,12 @@
       "type": "shell",
       "command": "EXCLUDE_CONTRIB=true tools/vscode/refresh_compdb.sh",
       "problemMatcher": []
+    },
+    {
+      "label": "Local Fix Format",
+      "type": "shell",
+      "command": "tools/vscode/local_fix_format.sh -all",
```
Comment (phlax) [inline .vscode/tasks.json:39]: i dont think this exists

Comment (phlax) [reply .vscode/tasks.json:39]: wrt local_fix_format im also a bit reluctant to hook this up officially as its not necessarily the best path to achieve this, but wont block - i dont use vscode myself

cc @ravenblackx @jmarantz

Comment (zirain) [reply .vscode/tasks.json]: my bad, write the wrong path.

**Comment by zirain** [comment]
kindly ping @phlax

**Comment by ravenblackx** [comment]
I should add, I don't know if it would work or be desirable to use a `bazel run` command in a menu option, which I think is probably what Phlax is thinking might be a better alternative than the direct script. A downside of `bazel run` is that bazel generally refuses to do two things at a time, so if it was done that way you couldn't e.g. reformat while a long slow build is in flight.

**Review by phlax — APPROVED** [review]
my hesitation was mostly that this code path is kinda unofficial - that said we do need something like it, so i think its reasonable to hook this (if its helpful to vscode users)

lgtm, thanks @zirain

**Pull request merged**